src/TLuaInterpreter.cpp

```diff
@@ -17968,154 +17968,72 @@ int TLuaInterpreter::getScroll(lua_State* L)
-        }
-#endif
-        lua_pushboolean(L, false);
-        return 1;
-    }
-
-    if (key == qsl("mapperPanelVisible")) {
-        lua_pushboolean(L, host.mShowPanel);
-        return 1;
-    }
-
-    if (key == qsl("mapShowRoomBorders")) {
-        lua_pushboolean(L, host.mMapperShowRoomBorders);
-        return 1;
-    }
-
-    if (key == qsl("enableGMCP")) {
-        lua_pushboolean(L, host.mEnableGMCP);
-        return 1;
-    }
-
-    if (key == qsl("enableMSDP")) {
-        lua_pushboolean(L, host.mEnableMSDP);
-        return 1;
-    }
-
-    if (key == qsl("enableMSSP")) {
-        lua_pushboolean(L, host.mEnableMSSP);
-        return 1;
-    }
-
-    if (key == qsl("enableMSP")) {
-        lua_pushboolean(L, host.mEnableMSP);
-        return 1;
-    }
-
-    if (key == qsl("askTlsAvailable")) {
-        lua_pushboolean(L, host.mAskTlsAvailable);
-        return 1;
-    }
-
-    if (key == qsl("inputLineStrictUnixEndings")) {
-        lua_pushboolean(L, host.mUSE_UNIX_EOL);
-        return 1;
-    }
-
-    if (key == qsl("autoClearInputLine")) {
-        lua_pushboolean(L, host.mAutoClearCommandLineAfterSend);
-        return 1;
-    }
-
-    if (key == qsl("showSentText")) {
-        lua_pushboolean(L, host.mPrintCommand);
-        return 1;
-    }
-
-    if (key == qsl("fixUnnecessaryLinebreaks")) {
-        lua_pushboolean(L, host.mUSE_IRE_DRIVER_BUGFIX);
-        return 1;
-    }
-
-    if (key == qsl("specialForceCompressionOff")) {
-        lua_pushboolean(L, host.mFORCE_NO_COMPRESSION);
-        return 1;
-    }
-
-    if (key == qsl("specialForceGAOff")) {
-        lua_pushboolean(L, host.mFORCE_GA_OFF);
-        return 1;
-    }
-
-    if (key == qsl("specialForceCharsetNegotiationOff")) {
-        lua_pushboolean(L, host.mFORCE_CHARSET_NEGOTIATION_OFF);
-        return 1;
-    }
-
-    if (key == qsl("specialForceMxpNegotiationOff")) {
-        lua_pushboolean(L, host.mFORCE_MXP_NEGOTIATION_OFF);
-        return 1;
-    }
-
-    if (key == qsl("compactInputLine")) {
-        lua_pushboolean(L, host.getCompactInputLine());
-        return 1;
-    }
-
-    if (key == qsl("announceIncomingText")) {
-        lua_pushboolean(L, host.mAnnounceIncomingText);
-        return 1;
-    }
-
-    if (key == qsl("blankLinesBehaviour")) {
-        const auto behaviour = host.mBlankLineBehaviour;
-        if (behaviour == Host::BlankLineBehaviour::Show) {
-            lua_pushstring(L, "show");
-        } else if (behaviour == Host::BlankLineBehaviour::Hide) {
-            lua_pushstring(L, "hide");
-        } else if (behaviour == Host::BlankLineBehaviour::ReplaceWithSpace) {
-            lua_pushstring(L, "replacewithspace");
-        }
-        return 1;
-    }
+        return warnArgumentValue(L, __func__, "you must provide a key");
+    }
+
+    static const std::unordered_map<QString, std::function<void(lua_State*, Host&)>> configMap = {
+        { "mapRoomSize", [](lua_State* L, Host& host){ lua_pushnumber(L, host.mRoomSize); } },
```

Comment:
All these "functionName" need to be `qsl(`...`)` wrapped...!

Reply:
Sure, done